File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Color Bricklet 2.0

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim c As New BrickletColorV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current color
        Dim r, g, b, c_ As Integer

        c.GetColor(r, g, b, c_)

        Console.WriteLine("Color [R]: " + r.ToString())
        Console.WriteLine("Color [G]: " + g.ToString())
        Console.WriteLine("Color [B]: " + b.ToString())
        Console.WriteLine("Color [C]: " + c_.ToString())

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/color_v2_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Color Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	c, _ := color_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current color.
	r, g, b, c_, _ := c.GetColor()

	fmt.Printf("Color [R]: %d\n", r)
	fmt.Printf("Color [G]: %d\n", g)
	fmt.Printf("Color [B]: %d\n", b)
	fmt.Printf("Color [C]: %d\n", c_)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
